### PR TITLE
Encapsulate several top-level menus

### PR DESCRIPTION
**Pull request by RussKie**
## Proposed changes

Encapsulate "Start", "Tools" and "Help" menus, thus reducing the complexity of the `FormBrowse` dialog. 

Why? I'm currently working on a POC that introduces a shell form that hosts different UI - such as the dashboard, the repo browser or the repo history views, as well as introduces dependency injection and UI composition. That work is extremely complex, and having encapsulated "start menu" item helps to avoiding code duplication and significantly reduces associated complexity.

## Test methodology 

- existing tests
- manual tests to verify translations are correctly applied

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

**Comment by mstv** [comment]
The last commit does not compile.
I am going to use this for the "Copy path(s)" context menu.

File: UnitTests/GitUI.Tests/RepositoryHistoryUIServiceTests.cs [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+using FluentAssertions;
+using GitCommands.UserRepositoryHistory;
+using GitUI;
+using GitUI.CommandsDialogs;
+using NSubstitute;
+using NUnit.Framework;
+
+namespace GitUITests
+{
+    [TestFixture]
```
Comment (mstv) [inline UnitTests/GitUI.Tests/RepositoryHistoryUIServiceTests.cs:11]:
```suggestion
    [Apartment(ApartmentState.STA)]
    [TestFixture]
```

**Comment by mstv** [comment]
It would have been good to append " (#10647)" to the commit messages.

**Comment by RussKie** [comment]
Ah, that's GitHub's doing. I think, it appends the commit's number when squash-merging, and not for rebase-merge.

**Comment by pmiossec** [comment]
@RussKie Can you investigate the refactoring around `RepositoryHistoryUIService` because this PR breaks the switch of repositories use "recent/pinned" repos?

switch_broken


We end up with `GitModuleChanged` being null and so switch is not done. I don't know if it's because there are 2 instances of `RepositoryHistoryUIService`...
image

**Comment by RussKie** [comment]
Sure thing. Thanks for letting me know.

**Pull request merged**